signer/tests/integration/setup.rs

```diff
@@ -0,0 +1,351 @@
+use bitcoin::consensus::Encodable as _;
+use bitcoin::hashes::Hash as _;
+use bitcoin::AddressType;
+use bitcoin::BlockHash;
+use bitcoin::OutPoint;
+use bitcoincore_rpc::Client;
+use bitcoincore_rpc::RpcApi as _;
+use blockstack_lib::types::chainstate::StacksAddress;
+use clarity::vm::types::PrincipalData;
+
+use fake::Fake;
+use fake::Faker;
+use rand::rngs::OsRng;
+use sbtc::testing::regtest;
+use sbtc::testing::regtest::Faucet;
+use sbtc::testing::regtest::Recipient;
+use signer::bitcoin::rpc::BitcoinCoreClient;
+use signer::bitcoin::rpc::BitcoinTxInfo;
+use signer::bitcoin::utxo;
+use signer::bitcoin::utxo::SbtcRequests;
+use signer::bitcoin::utxo::SignerBtcState;
+use signer::bitcoin::utxo::SignerUtxo;
+use signer::config::Settings;
+use signer::keys::PublicKey;
+use signer::keys::SignerScriptPubKey;
+use signer::storage::model;
+use signer::storage::model::BitcoinTxRef;
+use signer::storage::model::EncryptedDkgShares;
+use signer::storage::postgres::PgStore;
+use signer::storage::DbWrite as _;
+use signer::testing;
+
+use crate::utxo_construction::generate_withdrawal;
+use crate::utxo_construction::make_deposit_request;
+
+/// A struct containing an actual deposit and a sweep transaction. The
+/// sweep transaction was signed with the `signer` field's public key.
+pub struct TestSweepSetup {
+    /// The block hash of the bitcoin block that confirms the deposit
+    /// transaction.
+    pub deposit_block_hash: bitcoin::BlockHash,
+    /// Where the corresponding sBTC will be minted.
+    pub deposit_recipient: PrincipalData,
+    /// The deposit request, and a bitmap for how the singers voted on it.
+    pub deposit_request: utxo::DepositRequest,
+    /// The bitcoin transaction that the user made as a deposit for sBTC.
+    pub deposit_tx: bitcoin::Transaction,
+    /// The signer object. It's public key represents the group of signers'
+    /// public key, allowing us to abstract away the fact that there are
+    /// many signers needed to sign a transaction.
+    pub signer: Recipient,
+    /// The public keys of the signer set. It is effectively controlled by the above signer's private key.
+    pub signer_keys: Vec<PublicKey>,
+    /// The block hash of the bitcoin block that confirmed the sweep
+    /// transaction.
+    pub sweep_block_hash: bitcoin::BlockHash,
+    /// The height of the bitcoin block that confirmed the sweep
+    /// transaction.
+    pub sweep_block_height: u64,
+    /// The transaction that swept in the deposit transaction.
+    pub sweep_tx_info: BitcoinTxInfo,
+    /// The withdrawal request, and a bitmap for how the siggners voted on
+    /// it.
+    pub withdrawal_request: utxo::WithdrawalRequest,
+    /// The address that initiated with withdrawal request.
+    pub withdrawal_sender: PrincipalData,
+}
+
+impl TestSweepSetup {
+    /// Construct a new TestDepositSetup
```

Comment:
```suggestion
    /// Construct a new TestSweepSetup
```

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/582/commits/21ed295d7d378bea8d0eecf8e68031cd070b0e13